File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/SkinDao.java
package com.legue.axel.lolsummonertool.database.dao.champion;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.champion.Skin;

import java.util.List;

@Dao
public interface SkinDao {

    @Query("SELECT * FROM skins ORDER BY id")
    LiveData<List<Skin>> getSkins();

    @Query("SELECT * FROM skins WHERE id = :skinId")
    LiveData<Skin> getSkinById(int skinId);

    @Insert
    void insertSkin(Skin skin);

    @Delete
    void deleteSkin(Skin skin);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateSkin(Skin skin);
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/SkinDao.kt
package com.legue.axel.lolsummonertool.database.dao.champion

import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.champion.Skin

@Dao
interface SkinDao {
    @Query("SELECT * FROM skins ORDER BY id")
    fun getSkins(): LiveData<List<Skin>>

    @Query("SELECT * FROM skins WHERE id = :skinId")
    fun getSkinById(skinId: Int): LiveData<Skin>

    @Insert
    fun insertSkin(skin: Skin)

    @Delete
    fun deleteSkin(skin: Skin)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateSkin(skin: Skin)
}